services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/volume_files.py

```diff
@@ -0,0 +1,139 @@
+import os
+import stat
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional, Union
+
+import aiofiles
+from aiofiles import os as aiofiles_os
+from models_library.volumes import VolumeID
+from servicelib.volumes_utils import (
+    AGENT_FILE_NAME,
+    HIDDEN_FILE_NAME,
+    VolumeState,
+    save_volume_state,
+)
+
+from .mounted_fs import MountedVolumes
+
+chmod = aiofiles_os.wrap(os.chmod)  # type: ignore
+
+
+@dataclass
+class _MountedVolumesLocalPaths:
+    outputs: Path
+    inputs: Path
+    states: tuple[Path, ...]
+    shared_store: Path
+
+    @classmethod
+    def from_mounted_volumes(
+        cls, mounted_volumes: MountedVolumes
+    ) -> "_MountedVolumesLocalPaths":
+        return cls(
+            inputs=mounted_volumes.disk_inputs_path,
+            outputs=mounted_volumes.disk_outputs_path,
+            states=tuple(mounted_volumes.disk_state_paths()),
+            shared_store=mounted_volumes.disk_shared_store_path,
+        )
+
+    def paths_from_volume_id(self, volume_id: VolumeID) -> list[Path]:
+        result: Union[Path, tuple[Path, ...]] = self.__getattribute__(volume_id)
+        if isinstance(result, Path):
+            return [result]
+        return list(result)
+
+
+async def _create_file_with_restricted_permissions(file: Path) -> None:
+    """only allows the user who created the files to change it"""
+    # create empty file
+    async with aiofiles.open(file, mode="w"):
```

Comment:
I understand what you do but read very very strange having a with statment with ellipsis!

I wonder if we are not overdoing with async call and here we should just use
```python
file.touch( mode=(stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IWGRP | stat.S_IROTH), exist_ok=True)
```
would make the trick here and is pretty clear!!

Reply:
this looks very weird indeed. without your @pcrespov comment.. I was wondering what this is about... and I am pretty sure aiofiles might event have the async version of touch...

Reply:
So the only touch implementation is the one from pathlib.

@pcrespov fyi: last time I avoided used pathlib it took me days to find a subtle bug and I had to revert back to aiofiles. So I will figure out how to make pathlib's touch async.

Reply:
OK, using PC's suggestion seems to be working.